<?xml version="1.0"?>
<launch>
        <!-- <param name="robot_description" 
            command="$(find xacro)/xacro '$(find ppr_description)/urdf/ppr.xacro'"/> -->
    <group ns="motors">
        <rosparam command="load" file="$(find motor_driver)/config/port.yaml"/>

        <!-- Load hardware interface -->
        <node pkg = "motor_driver" name = "motor" type = "motor_driver" output="screen">
        </node>
        <!-- Load controller settings -->
        <rosparam command="load" file="$(find motor_driver)/config/controllers.yaml"/>

        <!-- Load controller manager -->
        <node name="controller_spawner" pkg="controller_manager" type="spawner"  output="screen" respawn="false" respawn_delay="0"
        args="joint_state_controller
                    joint1_position_controller
                    joint2_position_controller
                    joint3_position_controller"/>
                    
        <!-- Convert joint states to /tf tranforms -->
        <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

    </group>
</launch>